ChallengesWithTestsMarkVII/ChallengesSet03.cs

```diff
@@ -1,53 +1,81 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace ChallengesWithTestsMarkVII
 {
     public class ChallengesSet03
     {
         public bool ArrayContainsAFalse(bool[] vals)
         {
-            throw new NotImplementedException();
+            if (vals.Contains(false))
+            {
+                return true;
+            }
+            return false;
         }
 
         public bool IsSumOfOddsOdd(IEnumerable<int> numbers)
         {
-            throw new NotImplementedException();
+            if (numbers != null && numbers.Sum() % 2 != 0)
+            {
+                return true;
+            }
+            return false;
         }
 
         public bool PasswordContainsUpperLowerAndNumber(string password)
         {
-            throw new NotImplementedException();
+            if (password.Any(char.IsUpper) && password.Any(char.IsLower) && password.Any(char.IsDigit))
+            {
+                return true;
+            }
+            return false;
         }
 
         public char GetFirstLetterOfString(string val)
         {
-            throw new NotImplementedException();
+            return val[0];
         }
 
         public char GetLastLetterOfString(string val)
         {
-            throw new NotImplementedException();
+            return val[val.Length - 1];
         }
 
         public decimal Divide(decimal dividend, decimal divisor)
         {
-            throw new NotImplementedException();
+            if(divisor == 0)
+            {
+                return 0;
+            }
+            return dividend / divisor;
         }
 
         public int LastMinusFirst(int[] nums)
         {
-            throw new NotImplementedException();
+            return nums[nums.Length - 1] - nums[0] ;            
         }
 
         public int[] GetOddsBelow100()
         {
-            throw new NotImplementedException();
+            int index = 0;
+            int[] odds = new int[50];
+            for (int n = 1; n < (99 + 1); n++)
```

Comment:
This works, but consider changing up the for-loop to something that skips all the evens, something like `for (int i = 1; i < 100; i += 2)`. This will let you get rid of the if-statement inside the loop since `i` will increase by 2 each time around (1, 3, 5, 7, etc.)